client/components/shared/NewBankForm.js

```diff
@@ -119,72 +120,77 @@ export default class NewBankForm extends React.Component {
             );
 
             let maybeCustomFields = [];
-            if (this.state.hasCustomFields) {
+            if (this.state.customFields.length > 0) {
                 maybeCustomFields = this.state.customFields.map((field, index) =>
-                    <CustomBankField ref={ `customField${index}` } params={ field } />
+                    <CustomBankField ref={ `customField${index}${this.state.bankUuid}` } params={ field }
+                      key={ `customField${index}${this.state.bankUuid}` }
+                    />
                 );
             } else {
                 maybeCustomFields = <div/>;
             }
 
-            maybeForm = <div className="panel-body transition-expand">
-                <div className="form-group">
-                    <label htmlFor="bank">
-                        { $t('client.settings.bank') }
-                    </label>
-                    <select className="form-control" id="bank" ref="bank"
-                      onChange={ this.onChangedBank.bind(this) }>
-                        { options }
-                    </select>
-                </div>
+            maybeForm = (
+                <div className="panel-body transition-expand">
+                    <div className="form-group">
+                        <label htmlFor="bank">
+                            { $t('client.settings.bank') }
+                        </label>
+                        <select className="form-control" id="bank" ref="bank"
+                          onChange={ this.handleChangeBank }
+                          defaultValue={ this.state.bankUuid }>
```

Comment:
Nevermind this comment, I've looked at your PR message and I assume this is what shows the same selected bank when toggling the panel on/off.